pkg/apis/storage/types.go

```diff
@@ -203,6 +203,10 @@ type VolumeError struct {
 	// information.
 	// +optional
 	Message string
+
+	// ResourceExhaust indicates that the maximum supported number of volumes that can be attached to the target node are already attached.
+	// +optional
+	ResourceExhaust *bool
```

Comment:
Should we directly store grcp error code here or only a subset of error codes as enum or boolean? 

I am kinda leaning towards - 2nd option. Store attach error codes as enum and we can expand these if needed. @msau42 what do you think?

Reply:
Great idea, i've updated this field to an enum type that represents the different error codes listed for the ControllerPublishVolume RPC [here](https://github.com/container-storage-interface/spec/blob/master/spec.md#controllerpublishvolume-errors), this way we can expand if needed.